ouroboros-network/src/Ouroboros/Network/Diffusion/Common.hs

```diff
@@ -170,16 +172,18 @@ data Applications ntnAddr ntnVersion ntnVersionData
         :: (PeerSharingAmount -> m [ntnAddr])
         -> Versions ntnVersion
                     ntnVersionData
-                    (OuroborosBundle
+                    (OuroborosBundleWithExpandedCtx
                       InitiatorResponderMode ntnAddr
                       ByteString m a ())
 
       -- | NodeToClient responder application (server role)
       --
+      -- Because p2p mode does not infect local connections we we use non-p2p
+      -- apps.
     , daLocalResponderApplication
         :: Versions ntcVersion
                     ntcVersionData
-                    (OuroborosApplication
+                     (OuroborosApplicationWithMinimalCtx
                       ResponderMode ntcAddr
                       ByteString m Void ())
```

Comment:
We could use:
```
        :: forall ctx. Versions ntcVersion
                    ntcVersionData
                     (OuroborosApplication
                      ResponderMode ctx (ResponderContext ntcAddr)
                      ByteString m Void ())
```

Reply:
We cannot use an existential type here, one will not be able to provide a concrete choice.